components/user/EventRow.tsx

```diff
@@ -0,0 +1,297 @@
+    // https://github.com/iron-fish/ironfish/issues/930
+    return <>View issue #{id}</>
+  } else if (type === EventType.COMMUNITY_CONTRIBUTION) {
+    return <>View contribution</>
+  } else if (type === EventType.SOCIAL_MEDIA_PROMOTION) {
+    return <>Promoted on {hostname}</>
+  }
+  return <>UNHANDLED: {type}</>
+}
+
+const formatEventDate = (d: Date) =>
+  format(d, `y'-'MM'-'dd HH':'mm':'ss`, {
+    locale: enUS,
+  })
+
+export const EventRow = ({
+  type,
+  occurred_at: occurredAt,
+  points,
+  metadata,
+}: EventRowProps) => {
+  const { text, icon } = displayEventType(type)
+
+  const eType = (
+    <div className="flex items-center justify-start">
+      <span className="mr-2">{icon}</span>
+      {text}
+    </div>
+  )
+
+  return (
+    <tr className="border-b border-black">
+      <td className="py-4 w-1/3">{eType}</td>
+      <td>{formatEventDate(new Date(occurredAt))}</td>
+      <td>{points}</td>
+      <td className="max-w-[11rem]">
+        <a
+          href={makeLinkForEvent(type, metadata)}
+          className="text-ifotherblue align-left flex items-end justify-between"
+          target="_blank"
+          rel="noreferrer"
+        >
+          <div>{metadata && summarizeEvent(type, metadata)}</div>
+          <ChevronRight />
+        </a>
+      </td>
+    </tr>
+  )
+}
+
+type WeekRowProps = {
+  date: Date
+  week: number
+}
+
+const WeekRow = ({ date, week }: WeekRowProps) => {
+  const when = `Week ${week} - Started ${format(date, 'MMMM do, Y')}`
+  return (
+    <tr
+      className="bg-black text-white"
+      data-date={date}
+      aria-label={when}
+      title={when}
+    >
+      <td
+        colSpan={4}
+        className="text-center uppercase text-xs tracking-widest h-8"
+      >
+        Week {week}
+      </td>
+    </tr>
+  )
+}
+const weeksBetween = (start: Date, end: Date) =>
+  eachWeekOfInterval({ start, end })
+
+const eventsBetween = (
+  start: Date,
+  end: Date,
+  events: ApiEvent[]
+): ApiEvent[] =>
+  events.filter(e => {
+    const time = parseISO(e.occurred_at)
+    return isAfter(time, start) && isBefore(time, end)
+  })
+
+const makeCounter = () => {
+  let x = 0
+  return () => ++x
+}
+
+const sortEventsByDate = (xs: ApiEvent[]) =>
+  xs.sort((a: ApiEvent, b: ApiEvent): number =>
+    a.occurred_at > b.occurred_at ? -1 : 1
+  )
+
+type WeeklyData = {
+  week: number
+  date: Date
+  prior: Date
+  events: ApiEvent[]
+}
+
+export const renderEvents = (start: Date, rawEvents: readonly ApiEvent[]) => {
+  const now = new Date()
+  const weeks = weeksBetween(start, now)
+  const counter = makeCounter()
+  return (
+    weeks
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      .reduce((agg: any, date: Date) => {
+        const prev = agg[agg.length - 1]
+        const prior = prev ? prev.date : start
+        const events = sortEventsByDate(
+          eventsBetween(prior, date, rawEvents as ApiEvent[])
+        )
+        const week = counter()
```

Comment:
I haven't looked at it enough to figure out where the issue is, but it looks like the week count starts on 2 instead of 1, based on this user: https://website-testnet-4mat9grpo-ironfish.vercel.app/users/1

Reply:
Good catch. I addressed this.